kpops/components/base_components/kubernetes_app.py

```diff
@@ -21,19 +27,95 @@ class Config(CamelCaseConfig):
 
 
 # TODO: label and annotations
-
-
 class KubernetesApp(PipelineComponent):
     """Base kubernetes app"""
 
     _type = "kubernetes-app"
     app: KubernetesAppConfig
 
+    version: str | None = None
+
+    _helm_wrapper: Helm | None = None
+    _helm_diff: HelmDiff | None = None
+
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.check_compatible_name()
+        self.__check_compatible_name()
+
+    @property
+    def helm_wrapper(self) -> Helm:
+        if self._helm_wrapper is None:
+            self._helm_wrapper = Helm(self.config.helm_config)
+            helm_repo_config = self.get_helm_repo_config()
+            if helm_repo_config is not None:
+                self._helm_wrapper.add_repo(
+                    helm_repo_config.repository_name,
+                    helm_repo_config.url,
+                    helm_repo_config.repo_auth_flags,
+                )
+        return self._helm_wrapper
```

Comment:
perhaps you could consider using `cached_property`. then you don't need to do the caching manually using class vars

Reply:
Okay I changed it to this:
```python
    @cached_property
    def helm_wrapper(self) -> Helm:
        helm_wrapper = Helm(self.config.helm_config)
        helm_repo_config = self.get_helm_repo_config()
        if helm_repo_config is not None:
            helm_wrapper.add_repo(
                helm_repo_config.repository_name,
                helm_repo_config.url,
                helm_repo_config.repo_auth_flags,
            )
        return helm_wrapper
```

but I am getting the following error:
```console
tests/components/test_kubernetes_app.py:None (tests/components/test_kubernetes_app.py)
tests/components/test_kubernetes_app.py:15: in <module>
    from kpops.components.base_components.kubernetes_app import (
kpops/components/__init__.py:1: in <module>
    from kpops.components.base_components import (
kpops/components/base_components/__init__.py:1: in <module>
    from kpops.components.base_components.kafka_app import KafkaApp
kpops/components/base_components/kafka_app.py:8: in <module>
    from kpops.components.base_components.kubernetes_app import (
kpops/components/base_components/kubernetes_app.py:31: in <module>
    class KubernetesApp(PipelineComponent):
pydantic/main.py:222: in pydantic.main.ModelMetaclass.__new__
    ???
pydantic/fields.py:506: in pydantic.fields.ModelField.infer
    ???
pydantic/fields.py:436: in pydantic.fields.ModelField.__init__
    ???
pydantic/fields.py:546: in pydantic.fields.ModelField.prepare
    ???
pydantic/fields.py:570: in pydantic.fields.ModelField._set_default_and_type
    ???
pydantic/fields.py:439: in pydantic.fields.ModelField.get_default
    ???
pydantic/utils.py:695: in pydantic.utils.smart_deepcopy
    ???
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/copy.py:172: in deepcopy
    y = _reconstruct(x, memo, *rv)
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/copy.py:271: in _reconstruct
    state = deepcopy(state, memo)
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/copy.py:146: in deepcopy
    y = copier(x, memo)
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/copy.py:231: in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/copy.py:161: in deepcopy
    rv = reductor(4)
E   TypeError: cannot pickle '_thread.RLock' object
```

Reply:
It seems that they don't support it...
https://github.com/pydantic/pydantic/issues/1241

Reply:
ah too bad, typical pydantic headache 🙄
at least the workaround looks alright